lib/teamsnap/client.rb

```diff
@@ -0,0 +1,37 @@
+module TeamSnap
+  class Client
+    def initialize
+      yield(configuration) if block_given?
+    end
+
+    def method_missing(method, *args, &block)
+      #TODO: mutex/syncrhonize/just do once
+      TeamSnap.init(connection)
+      TeamSnap.const_get(Inflecto.camelize(method))
```

Comment:
@semmons99 @kyleries ideas on getting the connection into the run?

We know we want to avoid

```
TeamSnap::Team.find(1, connection)
```
But in this case we're returning a TeamSnap::Team which the application would then call `#find` on, so thinking about how to get the individual client's connection into there.